src/screens/MoreInfoScreen.js

```diff
@@ -1,19 +1,18 @@
-import React, { useState, useEffect } from 'react'
+import React, { useEffect } from 'react'
 import { Text, StyleSheet, ScrollView } from 'react-native'
 import ShowErrorMsg from '../components/ShowErrorMsg'
-import idApi  from '../hooks/useId'
+import useId  from '../hooks/useId'
 import ImageBox from '../components/ImageBox'
 import InfoBox from '../components/InfoBox'
 import ChipsBox from '../components/ChipsBox'
 
 
-const MoreInfoScreen = ({navigation}) => {
-  const restId = navigation.getParam('id')
-  const [result, setResult] = useState(null)
-  const [errorMsg, setErrorMsg] = useState('')
+const MoreInfoScreen = ({route}) => {
+  const {id} = route.params
+  const [idApi, result, errorMsg] = useId()
 
   useEffect(() => {
-    idApi(restId, setResult, setErrorMsg)
+    idApi(id)
   }, [])
```

Comment:
After a switch to the latest 'react-navigation', some things start to work differently, which caused to mistake in the MoreInfoScreen component. Add the 'id' variable to the 'route' object of 'navigation' and make access to it not directly but from the 'route' object inside the MoreInfoScreen.